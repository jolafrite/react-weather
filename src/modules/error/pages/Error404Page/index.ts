export * from './Error404Page';
export { default } from './Error404Page';
