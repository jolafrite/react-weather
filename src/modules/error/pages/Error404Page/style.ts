import { makeStyles, createStyles } from "@material-ui/core";
import { rem } from "polished";

export const useStyles = makeStyles(
  createStyles({
    root: {
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      height: '100%',
      width: '100%',
    },
    card: {
      minWidth: 275,
      padding: rem(10),
    },
    title: {
      textTransform: 'uppercase',
      marginBottom: rem(20),
      fontSize: rem(20),
      fontWeight: 'bold',
    },
    content: {

    },
    goToHomeLink: {
      display: 'flex',
      cursor: 'pointer',
      fontWeight: 'bolder',
      alignItems: 'center',
    }
  })
);
