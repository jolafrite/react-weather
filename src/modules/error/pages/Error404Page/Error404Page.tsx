import React, { FC, useCallback, memo } from "react";
import Layout from "../../../../common/components/Layout";
import { useStyles } from "./style";
import { Card, CardContent, CardActions, Link } from "@material-ui/core";
import { Typography } from '@material-ui/core';
import { push } from "connected-react-router";
import { useDispatch } from "react-redux";
import { KeyboardArrowLeft } from '@material-ui/icons';

export interface IError404PageProps {}

export const Error404Page: FC<IError404PageProps> = () => {
  const classes = useStyles();
  const dispatch = useDispatch();

  const goToHome = useCallback(
    () => dispatch(push('/', { prev: true }))
  , [dispatch]);

  return (
    <Layout>
      <div className={classes.root}>
        <Card className={classes.card}>
          <CardContent>
            <Typography
              className={classes.title}
              component="h1"
            >
              Page Not Found
            </Typography>

            <Typography
              className={classes.content}
              component="p"
            >
              Looks like you've followed a broken link or entered a URL
              that doesn't exist on this site.
            </Typography>
          </CardContent>
          <CardActions>
            <Link
              onClick={goToHome}
              className={classes.goToHomeLink}
            >
              <KeyboardArrowLeft />
              Back to home
            </Link>
          </CardActions>
        </Card>
      </div>
    </Layout>
  );
};

export default memo(Error404Page);
