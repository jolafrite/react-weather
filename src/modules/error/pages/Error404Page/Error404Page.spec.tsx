import React from 'react';
import { IError404PageProps, Error404Page } from "./Error404Page";
import { render, cleanup } from "@testing-library/react";
import { Provider } from 'react-redux';
import { createStore } from 'redux';

afterEach(cleanup);

describe('page: Error404Page', () => {

  const createComponent = (values = {}) => {
    const props: IError404PageProps = {
      ...values,
    };

    const store = createStore(() => {});
    return render(
      <Provider
        store={store}
      >
        <Error404Page {...props} />
      </Provider>
    );
  };

  test('default render', () => {
    const component = createComponent();

    expect(
      component.container
    ).toMatchSnapshot();
  });

});
