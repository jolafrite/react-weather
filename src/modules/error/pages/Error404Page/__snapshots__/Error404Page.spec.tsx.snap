// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page: Error404Page default render 1`] = `
<div>
  <div
    class="makeStyles-root-6"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation0 MuiAppBar-root MuiAppBar-positionSticky makeStyles-appBar-8 MuiAppBar-colorPrimary"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-9 MuiToolbar-gutters"
      >
        <div
          class="makeStyles-leftSlot-10"
        />
        <div
          class="makeStyles-brand-11"
        >
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineNone makeStyles-brandName-12 MuiTypography-h6 MuiTypography-colorInherit"
          >
            Weather App
          </a>
        </div>
        <div
          class="makeStyles-rightSlot-13"
        />
      </div>
    </header>
    <div
      class="makeStyles-container-7"
    >
      <div
        class="makeStyles-root-1"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-2 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          >
            <h1
              class="MuiTypography-root makeStyles-title-3 MuiTypography-body1"
            >
              Page Not Found
            </h1>
            <p
              class="MuiTypography-root makeStyles-content-4 MuiTypography-body1"
            >
              Looks like you've followed a broken link or entered a URL that doesn't exist on this site.
            </p>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing"
          >
            <a
              class="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-goToHomeLink-5 MuiTypography-colorPrimary"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
              </svg>
              Back to home
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
